tests/test_testclient.py

```diff
@@ -14,15 +25,54 @@ def mock_service_endpoint(request):
     return JSONResponse({"mock": "example"})
 
 
-def create_app(test_client_factory):
-    app = Starlette()
+_identity_runvar: anyio.lowlevel.RunVar[int] = anyio.lowlevel.RunVar("_identity_runvar")
+
+
+def get_identity(counter):
+    try:
+        return _identity_runvar.get()
+    except LookupError:
+        token = next(counter)
+        _identity_runvar.set(token)
+        return token
+
+
+def current_task():
+    # anyio's TaskInfo comparisons are invalid after their associated native
+    # task object is GC'd https://github.com/agronholm/anyio/issues/324
+    asynclib_name = sniffio.current_async_library()
+    if asynclib_name == "trio":
+        return trio.lowlevel.current_task()
+
+    if asynclib_name == "asyncio":
+        task = asyncio_current_task()
+        if task is None:
+            raise RuntimeError("must be called from a running task")  # pragma: no cover
+        return task
+    raise RuntimeError(f"unsupported asynclib={asynclib_name}")  # pragma: no cover
+
+
+def create_app(test_client_factory, counter=itertools.count()):
+    async def lifespan_context(app):
+        app.startup_task = current_task()
+        app.startup_loop = get_identity(counter)
```

Comment:
Since these attributes are only used in one test, would it make more sense to not modify this `create_app` function that is used by other tests and instead do this inside the test function? Could then use `nonlocal`s or something instead of assigning attributes to `app`.

Reply:
Yeah that makes sense, although I want to assign the taskgroup to app as a demo of how this would be used